### PR TITLE
Update SIG Azure configs

### PR DESCRIPTION
**Pull request by justaugustus**
- Update OWNERS
- Update plugins for cp-azure and capz
- Update capz presubmits
- Update [capz dashboards](https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure)

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

(I need to land updates to the capz tests first)
/hold

### CODE REVIEW

**Comment by justaugustus** [comment]
bazel for capz being [worked on](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/83), but tests will keep failing until I merge this.

/assign @feiskyer @tariq1890 @soggiest 
Guys, can you review this?

/hold cancel

File: prow/plugins.yaml [hunk 1]
```diff
@@ -356,6 +356,11 @@ plugins:
   - trigger
 
   kubernetes/cloud-provider-azure:
+  - milestone
+  - milestonestatus
```
Comment (spiffxp) [inline prow/plugins.yaml]: I won't hold for this, but this should go away.  Opened https://github.com/kubernetes/test-infra/issues/11037

Comment (BenTheElder) [reply prow/plugins.yaml]: we can remove it when we clean up #11037 😅

**Comment by tariq1890** [comment]
Ah I had missed to review this PR yesterday. Will do this shortly.

**Pull request merged**